# Concept Session: Working With Pandas

# MacPizza, one of the famous pizzerias in Unites States, has the data of its pizza sales for the last few years. The manager wants to analyze the past data for business growth, which is available in different file formats.

## Help the manager to perform various tasks to accomplish the goal.

# Task 1)

## The manager wants to read the data available in CSV file format. Help the manager to accomplish the task.

In [1]:
# Importing required library
import pandas as pd

In [2]:
# Reading data from csv file
data = pd.read_csv('bigmarket.csv')
data

Month Store  Sales
0     Jan     A  31037
1     Jan     B  20722
2     Jan     C  24557
3     Jan     D  34649
4     Jan     E  29795
5     Feb     A  29133
6     Feb     B  22695
7     Feb     C  28312
8     Feb     D  31454
9     Feb     E  46267
10  March     A  32961
11  March     B  26451
12  March     C  47814
13  March     D  36069
14  March     E  31874
15    Apr     A  27253
16    Apr     B  40241
17    Apr     C  47488
18    Apr     D  25432
19    Apr     E  33880
20    May     A  29487
21    May     B  40001
22    May     C  46482
23    May     D  46313
24    May     E  47594

In [3]:
data.shape

(25, 3)

## Interpretation:
### As you can see, the input data has 25 rows and 3 columns: Month, Store and Number of sales.

In [4]:
# Reading first five rows of input
data.head()

Month Store  Sales
0   Jan     A  31037
1   Jan     B  20722
2   Jan     C  24557
3   Jan     D  34649
4   Jan     E  29795

In [5]:
# Reading last five rows of input
data.tail()

Month Store  Sales
20   May     A  29487
21   May     B  40001
22   May     C  46482
23   May     D  46313
24   May     E  47594

# Task 2)

## The manager wants to read the data available in Excel file format. 

In [6]:
pwd

'e:\\'

In [7]:
!pip install openpyxl

You should consider upgrading via the 'C:\Users\ADMIN\AppData\Local\Programs\Python\Python310\python.exe -m pip install --upgrade pip' command.


In [8]:
# Reading data from Excel file
df = pd.read_excel('Bigmarket.xlsx')
df

Month Store  Sales
0     Jan     A  21230
1     Jan     B  32143
2     Jan     C  31977
3     Jan     D  45928
4     Jan     E  44129
5     Feb     A  41834
6     Feb     B  49442
7     Feb     C  24275
8     Feb     D  22811
9     Feb     E  23741
10  March     A  39763
11  March     B  34444
12  March     C  30385
13  March     D  33725
14  March     E  41534
15    Apr     A  30296
16    Apr     B  49449
17    Apr     C  37861
18    Apr     D  40768
19    Apr     E  29500
20    May     A  46617
21    May     B  37281
22    May     C  43326
23    May     D  43229
24    May     E  39154

# Task 3)

## The manager wants to read the data available in text file format. 

In [9]:
#'comma(,)' instead of space('\t') if there is comma's in b/w(instead of space)
#DF = pd.read_csv('big_market.txt', sep=',')

In [10]:
# Reading data from text file
DF = pd.read_csv('big_market.txt', sep='\t')
DF

Month Store  Sales
0     Jan     A  31037
1     Jan     B  20722
2     Jan     C  24557
3     Jan     D  34649
4     Jan     E  29795
5     Feb     A  29133
6     Feb     B  22695
7     Feb     C  28312
8     Feb     D  31454
9     Feb     E  46267
10  March     A  32961
11  March     B  26451
12  March     C  47814
13  March     D  36069
14  March     E  31874
15    Apr     A  27253
16    Apr     B  40241
17    Apr     C  47488
18    Apr     D  25432
19    Apr     E  33880
20    May     A  29487
21    May     B  40001
22    May     C  46482
23    May     D  46313
24    May     E  47594

# Task 4)

## The company has generated high revenue from the sales of pizzas online. Help the manager to read the data in HTML file format.

In [11]:
!pip install lxml

You should consider upgrading via the 'C:\Users\ADMIN\AppData\Local\Programs\Python\Python310\python.exe -m pip install --upgrade pip' command.


In [12]:
#Reading data from html file
DF1 = pd.read_html('Big_Market.html')
DF1

[    Unnamed: 0      A      B      C
 0            1  Month  Store  Sales
 1            2    Jan      A  31037
 2            3    Jan      B  20722
 3            4    Jan      C  24557
 4            5    Jan      D  34649
 5            6    Jan      E  29795
 6            7    Feb      A  29133
 7            8    Feb      B  22695
 8            9    Feb      C  28312
 9           10    Feb      D  31454
 10          11    Feb      E  46267
 11          12  March      A  32961
 12          13  March      B  26451
 13          14  March      C  47814
 14          15  March      D  36069
 15          16  March      E  31874
 16          17    Apr      A  27253
 17          18    Apr      B  40241
 18          19    Apr      C  47488
 19          20    Apr      D  25432
 20          21    Apr      E  33880
 21          22    May      A  29487
 22          23    May      B  40001
 23          24    May      C  46482
 24          25    May      D  46313
 25          26    May      E  47594]

# Task 5)

## a) Find the total number of rows and columns in input data. 

In [13]:
#Finding the number of rows and columns
df.shape

(25, 3)

### First entry indicates number of rows and second one indicates number of columns.

## b) Find the information about data types, shapes, and null values in input data.

In [14]:
#Finding the information about the data types, shape and null values of dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Month   25 non-null     object
 1   Store   25 non-null     object
 2   Sales   25 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 728.0+ bytes


In [15]:
df.describe() #recall excel!

Sales
count     25.000000
mean   36593.680000
std     8360.338927
min    21230.000000
25%    30385.000000
50%    37861.000000
75%    43229.000000
max    49449.000000

In [16]:
df['Sales'].describe()

count       25.000000
mean     36593.680000
std       8360.338927
min      21230.000000
25%      30385.000000
50%      37861.000000
75%      43229.000000
max      49449.000000
Name: Sales, dtype: float64

# Task 6)

## The manager wants to check few specific entries of data. 

In [17]:
# Extracting specific rows from dataframe 'df' 
df.iloc[11:16]

Month Store  Sales
11  March     B  34444
12  March     C  30385
13  March     D  33725
14  March     E  41534
15    Apr     A  30296

### .iloc: Integer-location based indexing for selection by position (from 0 to length-1 of the axis)
### Thus, [11:16] returns row numbers 11 to 16-1

# Task 7)

## The manager wants to check the entries for 'Store' only. 
### a) Method 1

In [18]:
# Extracting specific column
df.loc[20:24]['Store']
#df.loc[20:24,'Store'] alternative way
#

20    A
21    B
22    C
23    D
24    E
Name: Store, dtype: object

### The last five rows for 'Store' column have been displayed.
### b) Method 2

In [19]:
# Extracting specific column
df.iloc[20:25, 1] #tail or index number

20    A
21    B
22    C
23    D
24    E
Name: Store, dtype: object

# Task 8)

## The manager needs information of the stores where the sales are more than a specific number. 

In [20]:
# Find entry where sales are more than specific number
df[df.Sales > 45000]

Month Store  Sales
3    Jan     D  45928
6    Feb     B  49442
16   Apr     B  49449
20   May     A  46617

# The entries are extracted where sales are more than 45000.

# Task 9)

## The manager needs information of stores where the sales are less than a specific number. Help the manager to accomplish this for a particular month.

In [21]:
# Find entry where sales are less than specific number for particular month
x = df[(df.Month == 'Jan') & (df.Sales < 30000)]
x

Month Store  Sales
0   Jan     A  21230

### The entries are extracted where sales are less than 30000 in January.

# Task 10)

## The manager wants to offer bonus to the best stores to encourage high sales. He wants to select the top 5 stores with the highest sales record. 
### Sorting sales in descending order first

In [22]:
#Sorting values of sales in descending order
sort = df.sort_values('Sales', ascending=False)
sort

Month Store  Sales
16    Apr     B  49449
6     Feb     B  49442
20    May     A  46617
3     Jan     D  45928
4     Jan     E  44129
22    May     C  43326
23    May     D  43229
5     Feb     A  41834
14  March     E  41534
18    Apr     D  40768
10  March     A  39763
24    May     E  39154
17    Apr     C  37861
21    May     B  37281
11  March     B  34444
13  March     D  33725
1     Jan     B  32143
2     Jan     C  31977
12  March     C  30385
15    Apr     A  30296
19    Apr     E  29500
7     Feb     C  24275
9     Feb     E  23741
8     Feb     D  22811
0     Jan     A  21230

### Now, displaying the top 5 entries

In [23]:
#Displaying top 5 highest sales record
sort.iloc[:5, ] #also head( function)
#also sort[:5] 'slicing'

Month Store  Sales
16   Apr     B  49449
6    Feb     B  49442
20   May     A  46617
3    Jan     D  45928
4    Jan     E  44129

In [24]:
sort.head()

Month Store  Sales
16   Apr     B  49449
6    Feb     B  49442
20   May     A  46617
3    Jan     D  45928
4    Jan     E  44129

In [25]:
sort[:5]

Month Store  Sales
16   Apr     B  49449
6    Feb     B  49442
20   May     A  46617
3    Jan     D  45928
4    Jan     E  44129

# Task 11)

## The manager wants to access the sales in ascending order of 'Store' and 'Sales' values. How can the manager accomplish the task?

In [26]:
#Sorting values of Store and Sales in asscending order
df.sort_values(['Store','Sales'])

Month Store  Sales
0     Jan     A  21230
15    Apr     A  30296
10  March     A  39763
5     Feb     A  41834
20    May     A  46617
1     Jan     B  32143
11  March     B  34444
21    May     B  37281
6     Feb     B  49442
16    Apr     B  49449
7     Feb     C  24275
12  March     C  30385
2     Jan     C  31977
17    Apr     C  37861
22    May     C  43326
8     Feb     D  22811
13  March     D  33725
18    Apr     D  40768
23    May     D  43229
3     Jan     D  45928
9     Feb     E  23741
19    Apr     E  29500
24    May     E  39154
14  March     E  41534
4     Jan     E  44129

# Task 12) 

## The manager has found more records for a particular month. How can the manager add these new entries to the existing data?
### Method 1

In [27]:
# Adding new entires to input data
Dic = {'Month':['June','June','June','June','June'], 'Store':['A','B','C','D','E'], 'Sales':[28500,32896,30200,37098,41098]}
D = pd.DataFrame(Dic)
D

Month Store  Sales
0  June     A  28500
1  June     B  32896
2  June     C  30200
3  June     D  37098
4  June     E  41098

### New entries have been displayed

In [28]:
# Appending new entries to input
df = df.append(D,ignore_index=True)
df

C:\Users\ADMIN\AppData\Local\Temp\ipykernel_5028\1720626948.py:2: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(D,ignore_index=True)


Month Store  Sales
0     Jan     A  21230
1     Jan     B  32143
2     Jan     C  31977
3     Jan     D  45928
4     Jan     E  44129
5     Feb     A  41834
6     Feb     B  49442
7     Feb     C  24275
8     Feb     D  22811
9     Feb     E  23741
10  March     A  39763
11  March     B  34444
12  March     C  30385
13  March     D  33725
14  March     E  41534
15    Apr     A  30296
16    Apr     B  49449
17    Apr     C  37861
18    Apr     D  40768
19    Apr     E  29500
20    May     A  46617
21    May     B  37281
22    May     C  43326
23    May     D  43229
24    May     E  39154
25   June     A  28500
26   June     B  32896
27   June     C  30200
28   June     D  37098
29   June     E  41098

### Data has been displayed with the new entries added.

### Method 2

In [29]:
# Adding new entries
D1 = pd.concat([df,D],ignore_index=True)
D1

Month Store  Sales
0     Jan     A  21230
1     Jan     B  32143
2     Jan     C  31977
3     Jan     D  45928
4     Jan     E  44129
5     Feb     A  41834
6     Feb     B  49442
7     Feb     C  24275
8     Feb     D  22811
9     Feb     E  23741
10  March     A  39763
11  March     B  34444
12  March     C  30385
13  March     D  33725
14  March     E  41534
15    Apr     A  30296
16    Apr     B  49449
17    Apr     C  37861
18    Apr     D  40768
19    Apr     E  29500
20    May     A  46617
21    May     B  37281
22    May     C  43326
23    May     D  43229
24    May     E  39154
25   June     A  28500
26   June     B  32896
27   June     C  30200
28   June     D  37098
29   June     E  41098
30   June     A  28500
31   June     B  32896
32   June     C  30200
33   June     D  37098
34   June     E  41098

# Task 13)

## The company has its stores in five different states of US, namely 'Alaska', 'Texas', 'California', 'Montana', and 'New Mexico'. How can the manager add a new column that includes these five states?

In [30]:
#Creating data of 5 states
L = ['Alaska','Texas', 'California', 'Montana','New Mexico']    # Defining distinct values for new column
L1 = L*6                                                        # Create number of rows for this new column
Dic = {'State':L1}                                              # Define the name of column
df1 = pd.DataFrame(Dic)                                         # Create new column
df1                                                             # Display new column

State
0       Alaska
1        Texas
2   California
3      Montana
4   New Mexico
5       Alaska
6        Texas
7   California
8      Montana
9   New Mexico
10      Alaska
11       Texas
12  California
13     Montana
14  New Mexico
15      Alaska
16       Texas
17  California
18     Montana
19  New Mexico
20      Alaska
21       Texas
22  California
23     Montana
24  New Mexico
25      Alaska
26       Texas
27  California
28     Montana
29  New Mexico

### As you can see, a new column has been created, titled 'State', which has 30 rows and 5 distinct states.
### Now, let's add this new column to input data.

In [31]:
# Adding new column to input data
df = pd.concat([df,df1],axis=1)
df

Month Store  Sales       State
0     Jan     A  21230      Alaska
1     Jan     B  32143       Texas
2     Jan     C  31977  California
3     Jan     D  45928     Montana
4     Jan     E  44129  New Mexico
5     Feb     A  41834      Alaska
6     Feb     B  49442       Texas
7     Feb     C  24275  California
8     Feb     D  22811     Montana
9     Feb     E  23741  New Mexico
10  March     A  39763      Alaska
11  March     B  34444       Texas
12  March     C  30385  California
13  March     D  33725     Montana
14  March     E  41534  New Mexico
15    Apr     A  30296      Alaska
16    Apr     B  49449       Texas
17    Apr     C  37861  California
18    Apr     D  40768     Montana
19    Apr     E  29500  New Mexico
20    May     A  46617      Alaska
21    May     B  37281       Texas
22    May     C  43326  California
23    May     D  43229     Montana
24    May     E  39154  New Mexico
25   June     A  28500      Alaska
26   June     B  32896       Texas
27   June     C  30200  California
28   June     D  37098     Montana
29   June     E  41098  New Mexico

# Task 14)

## Help the manager to find the total sales of each state. Which state has the highest sales record?
### To find total sales in 'Alaska'

In [32]:
A = df[(df.State == 'Alaska')]                             # Find sales entries for Alaska
print("Total sales in Alaska is:",A['Sales'].sum())        # Sum of these entires

Total sales in Alaska is: 208240


In [33]:
A

Month Store  Sales   State
0     Jan     A  21230  Alaska
5     Feb     A  41834  Alaska
10  March     A  39763  Alaska
15    Apr     A  30296  Alaska
20    May     A  46617  Alaska
25   June     A  28500  Alaska

In [34]:
A['Sales'].std

<bound method NDFrame._add_numeric_operations.<locals>.std of 0     21230
5     41834
10    39763
15    30296
20    46617
25    28500
Name: Sales, dtype: int64>

In [35]:
A['Sales'].describe

<bound method NDFrame.describe of 0     21230
5     41834
10    39763
15    30296
20    46617
25    28500
Name: Sales, dtype: int64>

### To find total sales in 'Texas'

In [36]:
T = df[(df.State == 'Texas')]                             # Find sales entries for Texas
print("Total sales in Texas is:",T['Sales'].sum())        # Sum of these entires

Total sales in Texas is: 235655


### To find total sales in 'California'

In [37]:
C = df[(df.State == 'California')]                         # Find sales entries for California
print("Total sales in California is:",C['Sales'].sum())    # Sum of these entires

Total sales in California is: 198024


### To find total sales in 'Montana'

In [38]:
M = df[(df.State == 'Montana')]                            # Find sales entries for Montana
print("Total sales in Montana is:",M['Sales'].sum())       # Sum of these entires

Total sales in Montana is: 223559


### To find total sales in 'New Mexico'

In [39]:
N = df[(df.State == 'New Mexico')]                         # Find sales entries for New Mexico
print("Total sales in New Mexico is:",N['Sales'].sum())    # Sum of these entires

Total sales in New Mexico is: 219156


### To find the highest sales record

In [40]:
print("California has the highest sales record")

California has the highest sales record


# Task 15)

## Which steps should the manager follow to compare the performances of the two stores?
## Let's compare the sales of the two stores 'B' and 'D'.
### a) Extract and display the total number of sales for store B.

In [41]:
# Extract and display sales entries for store B
strB = df[df.Store=='B']
total_sale_B = strB.Sales.sum()
print("Total sales for Store B is:", total_sale_B)

Total sales for Store B is: 235655


### b) Extract and display the total number of sales for store D.

In [42]:
# Extract and display sales entries for store D
strD = df[df.Store=='D']
total_sale_D = strD.Sales.sum()
print("Total sales for Store D is:", total_sale_D)

Total sales for Store D is: 223559


### c) Now, find the difference between the sales of the two stores and respective percentages.

In [43]:
#percentage formula = " x2-x1/x1*100 "

In [44]:
x = total_sale_B - total_sale_D                   # Find the difference between two
y = x/total_sale_D                                # Find the percentage
print("Total sales for store D is %f %% less than the total sales for store D"%(y*100))

Total sales for store D is 5.410652 % less than the total sales for store D


### As you can see, the total sales for store B is 15.304963 % less than the total sales for store D.
### In this way, you can compare the sales of two stores.

# Task 16) 

## The manager wants to know the number of stores and states in the dataset. 

In [45]:
# Number of stores in data
df['Store'].value_counts()

A    6
B    6
C    6
D    6
E    6
Name: Store, dtype: int64

In [46]:
# Number of states in data
df['State'].value_counts()

Alaska        6
Texas         6
California    6
Montana       6
New Mexico    6
Name: State, dtype: int64

# Task 17)

## The manager wants to replace the values of the entries by some other values. How can the manager perform this task?
### Let's replace the sales value '32961' by a new value '33200'.

In [47]:
# Replace any value with some other value
df['Sales'] = df['Sales'].replace([21230],33200)
df

Month Store  Sales       State
0     Jan     A  33200      Alaska
1     Jan     B  32143       Texas
2     Jan     C  31977  California
3     Jan     D  45928     Montana
4     Jan     E  44129  New Mexico
5     Feb     A  41834      Alaska
6     Feb     B  49442       Texas
7     Feb     C  24275  California
8     Feb     D  22811     Montana
9     Feb     E  23741  New Mexico
10  March     A  39763      Alaska
11  March     B  34444       Texas
12  March     C  30385  California
13  March     D  33725     Montana
14  March     E  41534  New Mexico
15    Apr     A  30296      Alaska
16    Apr     B  49449       Texas
17    Apr     C  37861  California
18    Apr     D  40768     Montana
19    Apr     E  29500  New Mexico
20    May     A  46617      Alaska
21    May     B  37281       Texas
22    May     C  43326  California
23    May     D  43229     Montana
24    May     E  39154  New Mexico
25   June     A  28500      Alaska
26   June     B  32896       Texas
27   June     C  30200  California
28   June     D  37098     Montana
29   June     E  41098  New Mexico

# Task 18)

## The manager needs to update the record of the sales for a particular state. Help the manager to accomplish this task. 

### Let's update the record of 'Sales' in the state of Alaska for the month of March by the new sales value of '35000'.

### First, display all the records of Alaska state.

In [48]:
#Printing all the record of Alaska state
A

Month Store  Sales   State
0     Jan     A  21230  Alaska
5     Feb     A  41834  Alaska
10  March     A  39763  Alaska
15    Apr     A  30296  Alaska
20    May     A  46617  Alaska
25   June     A  28500  Alaska

### Then, update record of Alaska for March with new value '35000'.

In [49]:
#Updating sales record for the month of March
A.loc[A.Month == "March", 'Sales'] = 35000
A

Month Store  Sales   State
0     Jan     A  21230  Alaska
5     Feb     A  41834  Alaska
10  March     A  35000  Alaska
15    Apr     A  30296  Alaska
20    May     A  46617  Alaska
25   June     A  28500  Alaska

### Update the modified record in original input data.

In [50]:
df[(df.State == 'Alaska')] = A
df

Month Store  Sales       State
0     Jan     A  21230      Alaska
1     Jan     B  32143       Texas
2     Jan     C  31977  California
3     Jan     D  45928     Montana
4     Jan     E  44129  New Mexico
5     Feb     A  41834      Alaska
6     Feb     B  49442       Texas
7     Feb     C  24275  California
8     Feb     D  22811     Montana
9     Feb     E  23741  New Mexico
10  March     A  35000      Alaska
11  March     B  34444       Texas
12  March     C  30385  California
13  March     D  33725     Montana
14  March     E  41534  New Mexico
15    Apr     A  30296      Alaska
16    Apr     B  49449       Texas
17    Apr     C  37861  California
18    Apr     D  40768     Montana
19    Apr     E  29500  New Mexico
20    May     A  46617      Alaska
21    May     B  37281       Texas
22    May     C  43326  California
23    May     D  43229     Montana
24    May     E  39154  New Mexico
25   June     A  28500      Alaska
26   June     B  32896       Texas
27   June     C  30200  California
28   June     D  37098     Montana
29   June     E  41098  New Mexico

# Task 19)

## a) The manager wants to delete the initial 4 records from the dataset.
## b) Also, the manager wants to rename the column 'State' by the new name 'US_State'.
### a) To delete first 4 records from the dataset.

In [ ]:
#df3.drop('State',axis = 1,inplace=True)
#inplace = no need of variables it directly replaces the required conditions in the original data
#not recommended in the real world scenario's!
#***************************************
#df3.drop('State',axis = 1) 
#drop multiple columns drop([],axis=1)
#if axis = 0 search in the rows
#if axis = 1 search in the columns

In [51]:
df3 = df
df3 = df3.drop([0,1,2,3], axis=0)       # Dropping first 4 records
df3                                     # Print output

Month Store  Sales       State
4     Jan     E  44129  New Mexico
5     Feb     A  41834      Alaska
6     Feb     B  49442       Texas
7     Feb     C  24275  California
8     Feb     D  22811     Montana
9     Feb     E  23741  New Mexico
10  March     A  35000      Alaska
11  March     B  34444       Texas
12  March     C  30385  California
13  March     D  33725     Montana
14  March     E  41534  New Mexico
15    Apr     A  30296      Alaska
16    Apr     B  49449       Texas
17    Apr     C  37861  California
18    Apr     D  40768     Montana
19    Apr     E  29500  New Mexico
20    May     A  46617      Alaska
21    May     B  37281       Texas
22    May     C  43326  California
23    May     D  43229     Montana
24    May     E  39154  New Mexico
25   June     A  28500      Alaska
26   June     B  32896       Texas
27   June     C  30200  California
28   June     D  37098     Montana
29   June     E  41098  New Mexico

## b) To rename the column 'State' by the new name 'US_State'.

In [54]:
df = df.rename(columns={"State": "US_State"})
df

Month Store  Sales    US_State
0     Jan     A  21230      Alaska
1     Jan     B  32143       Texas
2     Jan     C  31977  California
3     Jan     D  45928     Montana
4     Jan     E  44129  New Mexico
5     Feb     A  41834      Alaska
6     Feb     B  49442       Texas
7     Feb     C  24275  California
8     Feb     D  22811     Montana
9     Feb     E  23741  New Mexico
10  March     A  35000      Alaska
11  March     B  34444       Texas
12  March     C  30385  California
13  March     D  33725     Montana
14  March     E  41534  New Mexico
15    Apr     A  30296      Alaska
16    Apr     B  49449       Texas
17    Apr     C  37861  California
18    Apr     D  40768     Montana
19    Apr     E  29500  New Mexico
20    May     A  46617      Alaska
21    May     B  37281       Texas
22    May     C  43326  California
23    May     D  43229     Montana
24    May     E  39154  New Mexico
25   June     A  28500      Alaska
26   June     B  32896       Texas
27   June     C  30200  California
28   June     D  37098     Montana
29   June     E  41098  New Mexico

# Task 20)

## The manager wants to know the state-wise sales record for the month of January. 

In [53]:
M1 = df[(df.Month == 'Jan')]     # Extract the entires for January
M1                               # Print output

Month Store  Sales    US_State
0   Jan     A  21230      Alaska
1   Jan     B  32143       Texas
2   Jan     C  31977  California
3   Jan     D  45928     Montana
4   Jan     E  44129  New Mexico